### PR TITLE
chore: Fix linter findings for prealloc (part3)

### PR DESCRIPTION
**Pull request by zak-pawel**
Address findings for [prealloc](https://github.com/alexkohler/prealloc) - Finds slice declarations that could potentially be pre-allocated.

This is the last part of addressing findings found by this linter, thus linter was also enabled in this PR.

related https://github.com/influxdata/telegraf/pull/12228 https://github.com/influxdata/telegraf/pull/12242

### CODE REVIEW

**Comment by zak-pawel** [comment]
!retry-failed

File: .golangci.yml [hunk 1]
```diff
@@ -76,6 +77,16 @@ linters-settings:
...
+  prealloc:
+    # Report pre-allocation suggestions only on simple loops that have no returns/breaks/continues/gotos in them.
+    # Default: true
+    simple: false
+    # Report pre-allocation suggestions on range loops.
+    # Default: true
+    range-loops: true
+    # Report pre-allocation suggestions on for loops.
+    # Default: false
+    for-loops: false
```
Comment (powersj) [inline .golangci.yml:89]: If these are defaults can we remove them?

**Review by powersj — COMMENTED** [review]
Thanks for the work on these! In order to avoid having a lengthy config file, can we remove the default settings you are adding? There are a couple of these already, but I do not believe we include them all.

File: .golangci.yml (hunk 1 above)
Comment (zak-pawel) [reply .golangci.yml]: @powersj Sure, done.

**Review by powersj — APPROVED** [review]

**Comment by zak-pawel** [comment]
!retry-failed

**Pull request merged**